impacket/smbserver.py

```diff
@@ -4413,7 +4413,7 @@ def processConfigFile(self, configFile = None):
 
         # Process the credentials
         credentials_fname = self.__serverConfig.get('global','credentials_file')
-        if credentials_fname is not "":
+        if credentials_fname != "":
```

Comment:
Referring to [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) "For sequences, (strings, lists, tuples), use the fact that empty sequences are false":
```suggestion
        if credentials_fname:
```